internal/cmd/signer.go

```diff
@@ -13,9 +13,21 @@ import (
 
 // GetSigner returns a signer according to the CLI context.
 func GetSigner(opts *SignerFlagOpts) (notation.Signer, error) {
+	// Check if using on-demand key
+	if opts.KeyID != "" && opts.PluginName != "" && opts.Key == "" {
+		// Construct a signer from on-demand key
+		mgr := plugin.NewCLIManager(dir.PluginFS())
+		plugin, err := mgr.Get(context.Background(), opts.PluginName)
+		if err != nil {
+			return nil, err
+		}
+		return signer.NewFromPlugin(plugin, opts.KeyID, map[string]string{})
```

Comment:
There is a pluginConfig in SignerFlagOpts

Reply:
I believe that pluginConfig is only in signOpts, not SignerFlagOpts.

Reply:
PluginConfig in `NewFromPlugin` is used only when we are reading keys which includes pluginConfig from singingkeys.json then this plugin config is then merged with pluginConfig passed via CLI in Sign and Verify operation to generate final pluginConfig.

Reply:
There is a --plugin-config flag for sign command

Reply:
Yes, it is one of the options passed to sign, but it doesn't get passed via the SignerFlagOpts. The pluginConfig that is used here is intended to be the one from the key, which is later merged with the values from the pluginConfig command line option you are speaking of. That's what Pritesh was referring to above.

Reply:
Sure, got it.

Reply:
@priteshbandi This is what I mentioned earlier in the comments of the original issue. A user might get confused on where did the key.PluginConfig go? (Although programmatically it works exactly the same, users have to remember to add those configs in `notation sign`'s pluginConfig now)